frontend/catalyst/pennylane_extensions.py

```diff
@@ -234,12 +243,18 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
+    methods = {"fd", "ps", "adj"}
     if method is None:
         method = "fd"
-    assert method in {"fd", "ps", "adj"}, "invalid differentiation method"
+    if method not in methods:
+        raise ValueError(
+            f"Invalid differentiation method '{method}'. "
+            f"Supported methods are: {' '.join(sorted(methods))}"
```

Comment:
Any reason for the sorting? 😅

Reply:
Just to keep output the same in case we reorder methods. Kind of increases output reproducibility.

Reply:
Not against it, was just wondering why you put it there :)